compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -612,6 +612,7 @@ prettyScenarioStep lvl (ScenarioStep stepId (Just step)) = do
     ScenarioStepStepCommit (ScenarioStep_Commit txId (Just tx) mbLoc) ->
       prettyCommit lvl txId mbLoc tx
 
+    -- Deprecated
```

Comment:
Would like to remove AssertMustFail, but it requires a change to the format in VSCode, which we currently test v1 and v2 script against.
We _could_ remove this and update the behaviour of v1 to match.
The behaviour change would be:
instead of steps being `commit + mustFail`, we get `commit + submissionFailed`. These steps are added regardless of if something goes wrong. e.g., currently, if a `mustFail` succeeds, the commit is added, but if a `mustSucceed` fails, nothing is added - I would like to add a `submissionFailed` there

Reply:
I would avoid changing v1 script. Since the problem here is that we use the same `*.EXPECTED.ledger` files for `script-v{1,2}`, perhaps we could split the `@LEDGER` annotations into `@LEDGER-V{1,2}`, then it would be a matter of search-replacing the existing annotations with pairs of these, duplicating the golden files and accepting any changes

Reply:
alternatively, we could add a `@SCRIPT-V1` annotation, duplicate the existing tests with `@LEDGER` annotations and associated `*.EXPECTED.ledger` files, then tag the originals with `@SCRIPT-V1` and the new ones with `-V2`. Something something adjunctions

Reply:
heh I just got to the part where you mention this, https://github.com/digital-asset/daml/pull/17974/files#r1425351963

Reply:
Yeah I think i'll extend the `Ledger` annotation to have a `LEDGER-V1`, `LEDGER-V2` and just `Ledger` for both
Or maybe `LEDGER` and `LEDGER-VERSIONED` which forces 2 paths, so you can't forget to provide a v2

Reply:
I started looking into this, and its a bigger change than I'd like. Many extra golden files, and probably some rewording in VSCode (currently all steps show under the title "Committed Transactions", which was never true, and now makes even less sense)
I will do this in a future PR [#18032](https://github.com/digital-asset/daml/issues/18032)